pkg/local_object_storage/metabase/metadata.go

```diff
@@ -0,0 +1,237 @@
+package meta
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+
+	objectcore "github.com/nspcc-dev/neofs-node/pkg/core/object"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/user"
+	"github.com/nspcc-dev/neofs-sdk-go/version"
+	"go.etcd.io/bbolt"
+)
+
+const (
+	metaPrefixA  = byte(iota)
+	metaPrefixBI // integer attributes
+	metaPrefixBS // all other attributes
+	metaPrefixC
+)
+
+var (
+	maxUint256 = new(big.Int).SetBytes([]byte{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255,
+		255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255})
+	maxUint256Neg = new(big.Int).Neg(maxUint256)
+)
+
+// TODO: fill on Init
+// TODO: system attributes
+func putMetadata(tx *bbolt.Tx, cnr cid.ID, id oid.ID, ver version.Version, owner user.ID, typ object.Type, creationEpoch uint64,
+	payloadLen uint64, pldHash, pldHmmHash, splitID []byte, parentID, firstID oid.ID, attrs []object.Attribute) error {
+	mb, err := tx.CreateBucketIfNotExists(metaBucketKey(cnr))
+	if err != nil {
+		return fmt.Errorf("create meta bucket for container: %w", err)
+	}
+	idk := [1 + oid.Size]byte{metaPrefixA}
+	copy(idk[1:], id[:])
+	if err := mb.Put(idk[:], nil); err != nil {
+		return fmt.Errorf("put object ID to container's meta bucket: %w", err)
+	}
+
+	var k []byte
+	// TODO: move to global funcs
+	makeKeyB := func(prefix byte, attr string, valLen int) (int, int) {
+		ln := 1 + oid.Size + len(attr) + valLen + len(utf8Delimiter)*2 // TODO: constantize some stuff
```

Comment:
`slices.Concat()`?

Reply:
yeah i thought about this. Will make code improvement after functionaly stabilization if u dont mind

Reply:
added key buffering, concatenation does not fit him